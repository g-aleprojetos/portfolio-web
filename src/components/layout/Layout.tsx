import React, {ReactNode} from 'react';
import {useBackgroundContext} from 'context/background';
import * as S from './Layout.styles';

type ILayout = {
  children: ReactNode;
};

export const Layout = ({children}: ILayout) => {
  const {themeDark} = useBackgroundContext();
  return (
    <S.Container data-testid={'layout-page'} data-backgroundblack={themeDark}>
      {children}
    </S.Container>
  );
};
