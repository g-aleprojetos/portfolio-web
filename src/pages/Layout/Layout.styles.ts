import styled from 'styled-components';
import {colors} from 'resources/colors';

export interface ExtraProps {
  $backgroundblack: boolean;
}

export const Container = styled.div<ExtraProps>`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  background-color: ${({$backgroundblack}) =>
    $backgroundblack ? colors.background02 : colors.darkgray};
`;

export const Content = styled.div`
  display: flex;
  position: fixed;
  width: 100%;
  height: 100%;
  margin-top: 116px;
  margin-bottom: 70px;
  overflow-y: auto;
`;
