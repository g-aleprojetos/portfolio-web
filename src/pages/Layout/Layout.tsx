import React from 'react';
import {Outlet, Link} from 'react-router-dom';
import {useBackgroundContext} from 'context/background';
import {ToggleSwitch} from 'components/toggleSwitch';
import {Dropdown} from 'components/dropDown';
import {useTranslation} from 'react-i18next';
import {namespaces} from 'utils/i18n/i18n.constants';
import routes from 'resources/routes';
import 'utils/i18n';
import * as S from './Layout.styles';

export const Layout = () => {
  const {themeDark, handleToggle} = useBackgroundContext();
  const {t} = useTranslation(namespaces.pages.header);
  return (
    <S.Container data-testid={'layout-page'} data-backgroundblack={themeDark}>
      <S.Header>
        <S.Nav>
          <S.Item>
            <S.ItemContent>
              <Link to={routes.Home}>
                <S.TextoHeader>{t('home')}</S.TextoHeader>
              </Link>
            </S.ItemContent>
          </S.Item>
          <S.Item>
            <S.ItemContent>
              <Link to={routes.Projects}>
                <S.TextoHeader>{t('contents')}</S.TextoHeader>
              </Link>
            </S.ItemContent>
          </S.Item>
          <S.Item>
            <S.ItemContent>
              <Link to={routes.About}>
                <S.TextoHeader>{t('about')}</S.TextoHeader>
              </Link>
            </S.ItemContent>
          </S.Item>
          <S.Item>
            <Dropdown />
          </S.Item>
          <S.Item>
            <ToggleSwitch isOn={themeDark} handleOnclick={handleToggle} />
          </S.Item>
        </S.Nav>
      </S.Header>
      <Outlet />
    </S.Container>
  );
};
