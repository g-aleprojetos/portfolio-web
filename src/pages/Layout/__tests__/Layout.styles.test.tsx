import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../Layout.styles';

describe('Layout.styles', () => {
  test('Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a true', () => {
    const {container} = render(<S.Container data-backgroundblack={true} />);
    expect(container).toMatchSnapshot();
  });

  test('Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a false', () => {
    const {container} = render(<S.Container data-backgroundblack={false} />);
    expect(container).toMatchSnapshot();
  });

  test('Item DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Item />);
    expect(container).toMatchSnapshot();
  });

  test('ItemContent DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.ItemContent />);
    expect(container).toMatchSnapshot();
  });

  test('Header DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Header />);
    expect(container).toMatchSnapshot();
  });

  test('Nav DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Nav />);
    expect(container).toMatchSnapshot();
  });

  test('TextoHeader DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextoHeader />);
    expect(container).toMatchSnapshot();
  });
});
