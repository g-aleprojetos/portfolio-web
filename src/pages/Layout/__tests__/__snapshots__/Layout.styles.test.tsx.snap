// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout.styles Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a false 1`] = `
.c0 {
  text-align: center;
  min-height: 100vh;
  background-color: #23242a;
}

<div>
  <div
    class="c0"
    data-backgroundblack="false"
  />
</div>
`;

exports[`Layout.styles Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a true 1`] = `
.c0 {
  text-align: center;
  min-height: 100vh;
  background-color: #F7F5F2;
}

<div>
  <div
    class="c0"
    data-backgroundblack="true"
  />
</div>
`;

exports[`Layout.styles Header DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Layout.styles Item DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Layout.styles ItemContent DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 10px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Layout.styles Nav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  padding-right: 30px;
  gap: 10px;
}

<div>
  <nav
    class="c0"
  />
</div>
`;

exports[`Layout.styles TextoHeader DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 15px;
  color: #fff;
  cursor: pointer;
}

<div>
  <p
    class="c0"
  />
</div>
`;
